lib/netext/httpext/request.go

```diff
@@ -115,8 +174,45 @@ func MakeRequest(ctx context.Context, preq *ParsedHTTPRequest) (*Response, error
 		Cookies: stdCookiesToHTTPRequestCookies(preq.Req.Cookies()),
 		Headers: preq.Req.Header,
 	}
+
+	if contentLength := preq.Req.Header.Get("Content-Length"); contentLength != "" {
+		length, err := strconv.Atoi(contentLength)
+		if err == nil {
+			preq.Req.ContentLength = int64(length)
+		}
+		// TODO: maybe do something in the other case ... but no error
+	}
+
 	if preq.Body != nil {
+		preq.Req.Body = ioutil.NopCloser(preq.Body)
+
+		// TODO: maybe hide this behind of flag in order for this to not happen for big post/puts?
+		// should we set this after the compression? what will be the point ?
 		respReq.Body = preq.Body.String()
+
+		switch {
+		case len(preq.Compressions) > 0:
+			compressedBody, length, contentEncoding, err := compressBody(preq.Compressions, preq.Req.Body)
+			if err != nil {
+				return nil, err
+			}
+
+			preq.Req.Body = ioutil.NopCloser(compressedBody)
+			if preq.Req.Header.Get("Content-Length") == "" {
+				preq.Req.ContentLength = length
+			} else {
+				state.Logger.Warningf(compressionHeaderOverwriteMessage, "Content-Length", preq.Req.Method, preq.Req.URL)
+			}
+			if preq.Req.Header.Get("Content-Encoding") == "" {
+				preq.Req.Header.Set("Content-Encoding", contentEncoding)
+			} else {
+				state.Logger.Warningf(compressionHeaderOverwriteMessage, "Content-Encoding", preq.Req.Method, preq.Req.URL)
+			}
+		case preq.Req.Header.Get("Content-Length") == "":
+			preq.Req.ContentLength = int64(preq.Body.Len())
+		default:
+			state.Logger.Warningf("Content-Length is specifically set won't reset it based on body length")
```

Comment:
As I mentioned in the previous comment, we probably should either remove this message (because current HAR converter scripts will trigger it on every POST/PUT/etc.), or we should at least add the URL in the error like the above messages

Reply:
I am going to remove it for now ... I think it will be best if we can somehow only emit messages for such things *once* per run but this is definitely not happing in this PR

Reply:
hmm yeah, that's a very good idea :+1: definitely a separate PR :grin:  but something we should investigate, since it will likely be a big UX win

implementation-wise, we could hash error messages and even count the number of times the same warning message would have been emitted... I remember seeing something like this in some sort of a CI or task-execution system or something like that that was done very well, but I can't remember the specifics unfortunately :disappointed: